### DISCUSSION TITLE
Persisting params on navigate

### DISCUSSION THREAD

**Discussion by barbinbrad** [remix-run/remix#5032] [Q&A]
Nested routing is wonderful, but it has some interesting problems. For example:

If I have this route, I'd normally be displaying some sort of table:

`http://host/users?search=foo&offset=50`

But suppose I want to edit a user in a drawer that overlays the table. When I navigate to:

`http://hosts/users/123`

My underlying table is repainted (because the params are gone).

Of course, I could _not_ use nested routing for the drawer. But I like nested routing. So what I've started doing is passing the params through like this:

```tsx
const [params] = useSerachParams();

<Link
   to={`123?${params.toString()}`
> Edit User</Link>

```

To be clear, I want to avoid the underlying table going back to it's default state when the drawer is shown. But it feels a bit verbose to concat the params onto the end of the URL. Any ideas on a better way to do this?

**Comment by kiliman** [comment]
That's just how URLs work. If you want to display information on the same page, you need to include everything that the page needs to render. In this case, since the filtered table and the user editor are all on the same page, you need to include all the params. 

The alternative is to maintain client state. That just adds complexity, and you lose some of the benefits of Remix.

Anyway, I have a helper `useScopedParams` that makes it easy to maintain params across pages.
 
https://github.com/kiliman/remix-scoped-params

**Reply by barbinbrad** [reply]
Hey thanks for the link and the response (and all the other stuff that you do around here). That makes sense. 

I was thinking about something like this for readability:

```tsx
<Link to="123" withParams>Edit User</Link>
```

For me, this is something I'd use everywhere, but I guess it depends on how you're architecting your routes/layouts. Guessing that's something that wouldn't make the cut for a feature PR?

**Reply by kiliman** [reply]
`<Link>` actually comes from React Router, so you'd need to make the PR there.

I don't have a problem with making my own customer wrapper components. It's unlikely they'll ever add every custom feature request to the core. I'll either do it in userland with custom components or patch the code as needed to work around issues or limitations.

Thanks for the kind words. Good luck with your project!
